File: CPP/test/Test_DogMeat.cpp
#include "../src/DogMeat.cpp"
#include <gtest/gtest.h>

using namespace std;

TEST(DogMeatTest, AllFunctionalityofDogMeat){
    DogMeat testDM(40, "Rexus");
    ASSERT_EQ("Rexus", testDM.getProducer());
    ASSERT_EQ(DOG_MEAT, testDM.getObjectType());
    ASSERT_EQ(FARMPRODUCT, testDM.getProductType());
    ASSERT_EQ(40, testDM.getPrice());
}

File: Java/src/test/java/engisubs/gameobject/product/farmproduct/meat/TestDogMeat.java
package engisubs.gameobject.product.farmproduct.meat;

import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;
import engisubs.gameobject.GameObject.GameObjectType;
import engisubs.gameobject.product.Product.ProductType;
public class TestDogMeat{
    private DogMeat test;
    @Before
    public void initObject(){
        test = new DogMeat(25, "Name");
    }

    @Test
    public void instantiationTest(){
        Assert.assertEquals("Name", test.getProducer());
        Assert.assertEquals(GameObjectType.DOG_MEAT, test.getObjectType());
        Assert.assertEquals(ProductType.FARMPRODUCT, test.getProductType());
        Assert.assertEquals(25, test.getPrice());
    }

}